library/std/src/collections/hash/set.rs

```diff
@@ -588,6 +588,11 @@ where
     /// Visits the values representing the intersection,
     /// i.e., the values that are both in `self` and `other`.
     ///
+    /// Note: When an equal element is present in `self` and `other`
+    /// then the resulting `Intersection` may yield references to
+    /// one or the other, which will be visible in properties of `T`
+    /// not participating in the `Eq` implementation.
```

Comment:
Saying it "will be visible" is maybe misleading because it won't necessarily be visible in this way, i.e. an arbitrary `T` doesn't necessarily have fields which don't participate in its `Eq` impl. And in the sense that it's *always* visible based on the memory address of the references produced by the iterator, that visibility occurs in a way that is not described by this sentence, and almost never relevant compared to the more common case of fields which don't participate in `Eq`.

Something like this would be clearer I think:

```suggestion
    /// one or the other. This can be relevant if `T` contains fields which
    /// are not compared by its `Eq` implementation, and may hold different
    /// value between the two equal copies of `T` in the two sets.
```